dotnet-efcore/OptimisticConcurrency/OptimisticConcurrency/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,139 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using OptimisticConcurrency.Data;
+using OptimisticConcurrency.Model;
+
+namespace OptimisticConcurrency.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class ProductsController : ControllerBase
+{
+    private const string ETagHeader = "ETag";
+    private const string IfMatchHeader = "If-Match";
+
+    private readonly ProductDbContext _context;
+
+    public ProductsController(ProductDbContext context)
+    {
+        _context = context;
+    }
+
+    [HttpGet]
+    public async Task<ActionResult<IEnumerable<Product>>> GetProduct()
+    {
+        if (_context.Product is null)
+        {
+            return NotFound();
+        }
+        return await _context.Product.ToListAsync();
+    }
+
+    [HttpGet("{id}")]
+    public async Task<ActionResult<Product>> GetProduct(int id)
+    {
+        if (_context.Product == null)
```

Comment:
For consistency in the code we should change these null checks to `is null` also.